runtime/runtime_test.go

```diff
@@ -4673,11 +4673,31 @@ func TestRuntimeRandom(t *testing.T) {
 		t.Parallel()
 
 		runValidCaseWithModulo := func(t *testing.T, ty sema.Type) {
+			byteSize := numericTypeByteSize(t, ty)
+
 			moduloBuffer := newRandBuffer(t)
 
+			// ensure random buffer range used for modulo is not all zeros (modulo cannot be zero)
+			for {
+				var foundNonZero bool
+				for _, v := range moduloBuffer[:byteSize] {
+					if v != 0 {
+						foundNonZero = true
+						break
+					}
+				}
+				if foundNonZero {
+					break
+				}
+
+				// if all zeros, draw new random buffer
+				_, err := rand.Read(moduloBuffer[:byteSize])
+				require.NoError(t, err)
+			}
+
 			// build a big Int from the modulo buffer, with the required `ty` size
 			// big.Int are used as they cover all the tested types including the small ones (UInt8 ..)
-			modulo := new(big.Int).SetBytes(moduloBuffer[:typeToBytes(t, ty)])
+			modulo := new(big.Int).SetBytes(moduloBuffer[:byteSize])
```

Comment:
Nit: we could use this `big.Int` to check for zero (using `Cmp`) instead of the buffer byte check.

Reply:
Good idea, could you please open a PR?

Reply:
yep, will clean it up in a few days.